components/eamxx/src/share/atm_process/atmosphere_process.cpp

```diff
@@ -1121,18 +1157,44 @@ ::remove_group (const std::string& group_name, const std::string& grid_name) {
   rmg(m_groups_out, m_groups_out_pointers);
 }
 
-void AtmosphereProcess::compute_column_conservation_checks_data (const int dt)
+void AtmosphereProcess::compute_column_conservation_checks_data (const double dt)
 {
-  EKAT_REQUIRE_MSG(m_column_conservation_check.second != nullptr,
+  EKAT_REQUIRE_MSG(m_conservation.second != nullptr,
                    "Error! User set enable_column_conservation_checks=true, "
-                   "but no conservation check exists.\n");
+                   "or has_energy_fixer=true, "
+                   "but no conservation check class exists.\n");
 
   // Set dt and compute current mass and energy.
   const auto& conservation_check =
-      std::dynamic_pointer_cast<MassAndEnergyColumnConservationCheck>(m_column_conservation_check.second);
+      std::dynamic_pointer_cast<MassAndEnergyColumnConservationCheck>(m_conservation.second);
   conservation_check->set_dt(dt);
   conservation_check->compute_current_mass();
   conservation_check->compute_current_energy();
 }
 
+void AtmosphereProcess::fix_energy (const double dt, const bool &print_debug_info)
+{
+  EKAT_REQUIRE_MSG(m_conservation.second != nullptr,
+                   "Error! User set has_energy_fixer=true, "
+                   "but no conservation check class exists.\n");
+
+  // Set dt and compute current mass and energy.
+  const auto& conservation_check =
+      std::dynamic_pointer_cast<MassAndEnergyColumnConservationCheck>(m_conservation.second);
+
+  //dt is needed to convert flux to change
+  conservation_check->set_dt(dt);
+  conservation_check->global_fixer(print_debug_info);
+
+  if(print_debug_info){
+    //print everything about the fixer only in debug mode
+    m_atm_logger->info("EAMxx:: energy fixer, PB T tendency " + std::to_string( conservation_check->get_pb_fixer() ) );
```

Comment:
"PB T" won't mean much to others. Perhaps something like "T tendency added to each physics cell"?

Reply:
done thanks